clippy_lints/src/duration_suboptimal_units.rs

```diff
@@ -0,0 +1,181 @@
+use clippy_config::Conf;
+use clippy_utils::consts::{ConstEvalCtxt, Constant};
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::msrvs::{self, Msrv};
+use clippy_utils::res::MaybeDef;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, QPath, RustcVersion};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::impl_lint_pass;
+use rustc_span::symbol::sym;
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Checks for instances where a `std::time::Duration` is constructed using a smaller time unit
+    /// when the value could be expressed more clearly using a larger unit.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Using a smaller unit for a duration that is evenly divisible by a larger unit reduces
+    /// readability. Readers have to mentally convert values, which can be error-prone and makes
+    /// the code less clear.
+    ///
+    /// ### Example
+    /// ```
+    /// use std::time::Duration;
+    ///
+    /// let dur = Duration::from_millis(5_000);
+    /// let dur = Duration::from_secs(180);
+    /// let dur = Duration::from_mins(10 * 60);
+    /// ```
+    ///
+    /// Use instead:
+    /// ```
+    /// use std::time::Duration;
+    ///
+    /// let dur = Duration::from_secs(5);
+    /// let dur = Duration::from_mins(3);
+    /// let dur = Duration::from_hours(10);
+    /// ```
+    #[clippy::version = "1.94.0"]
+    pub DURATION_SUBOPTIMAL_UNITS,
+    pedantic,
+    "constructing a `Duration` using a smaller unit when a larger unit would be more readable"
+}
+
+impl_lint_pass!(DurationSuboptimalUnits => [DURATION_SUBOPTIMAL_UNITS]);
+
+pub struct DurationSuboptimalUnits {
+    msrv: Msrv,
+}
+
+impl DurationSuboptimalUnits {
+    pub fn new(conf: &'static Conf) -> Self {
+        Self { msrv: conf.msrv }
+    }
```

Comment:
Others suggestion would lead to something like:

```suggestion
    pub fn new(tcx: TyCtxt<'_>, conf: &'static Conf) -> Self {
        let mut units = UNITS.to_vec();
        if tcx.features().enabled(sym::duration_constructors) {
            units.extend(EXTENDED_UNITS.iter().cloned());
        }
        Self { msrv: conf.msrv, units }
    }
```

Reply:
You are absolutely right, now simply went with `units.extend(EXTENDED_UNITS);`